cfme/infrastructure/datastore.py

```diff
@@ -241,15 +268,13 @@ def get_vms(self):
             view.entities.relationships.click_at('VMs')
         else:
             view.entities.relationships.click_at('Managed VMs')
-        # todo: to replace with correct view
-        vms_view = view.browser.create_view(DatastoresView)
-        return [vm.name for vm in vms_view.entities.get_all()]
+        vms_view = view.browser.create_view(DatastoreManagedVMsView)
+        return vms_view.entities.all_entity_names
 
     def delete_all_attached_vms(self):
         view = navigate_to(self, 'Details')
         view.entities.relationships.click_at('Managed VMs')
-        # todo: to replace with correct view
-        vms_view = view.browser.create_view(DatastoresView)
+        vms_view = view.browser.create_view(DatastoreManagedVMsView)
```

Comment:
I would register on navigation destination against `Datastore` for `ManagedVMs`

Reply:
ahhhh... we have already navigation destination for `Managed VMS`. I would to refactor this in same PR. 
https://github.com/ManageIQ/integration_tests/blob/65614529eb7543faf4d65c8128e181ac36088256/cfme/infrastructure/datastore.py#L468